extensions/auth/portability-auth-flickr/src/main/java/org/datatransferproject/auth/flickr/FlickrOAuthConfig.java

```diff
@@ -49,17 +53,28 @@ public String getAccessTokenUrl() {
   }
 
   @Override
-  public Map<String, String> getExportScopes() {
-    return ImmutableMap.of("PHOTOS", "read");
+  public List<String> getExportTypes() {
+    return ImmutableList.of("PHOTOS");
   }
 
   @Override
-  public Map<String, String> getImportScopes() {
-    return ImmutableMap.of("PHOTOS", "write");
+  public List<String> getImportTypes() {
+    return ImmutableList.of("PHOTOS");
   }
 
-  @Override
-  public String getScopeParameterName() {
-    return "perms";
+  public Map<String, String> getAdditionalUrlParameters(
+      String dataType, AuthMode mode, OAuth1Step step) {
+    if (dataType.equals("PHOTOS")) {
+      if (step == OAuth1Step.AUTHORIZATION) {
+        if (mode == AuthMode.EXPORT) {
+          return ImmutableMap.of(PERMS, "read");
+        } else {
+          return ImmutableMap.of(PERMS, "write");
+        }
+      }
```

Comment:
nit: I think it would be cleaner if you combined the step and mode conditions in the same if statement instead of nesting them 

OR the other thing you could do is create a static multimap, then you could do something like urlparams.get(datatype).get(mode).get(step) here -- this might be overkill for these two configs since they only support 1 datatype each but just a thought.

Reply:
I started with a static map and then decided it was overkill. Lots of extra extra code and repeated data. I'll simplify the steps.